src/reducers/ChatDux.js

```diff
@@ -23,19 +22,14 @@ const chat = createSlice({
       state.chatConversation = payload;
     },
     addNewMessage: (state, { payload }) => {
-      const { newChat, chatRoomId } = payload;
+      const { newChat, chatRoomId, updatedAt } = payload;
       state.chatConversation.conversation.push({ ...newChat });
-      const index = _findIndex(
-        state.chatRooms,
-        c => c.chatRoomId === chatRoomId
-      );
-      state.chatRooms.splice(index, 1, payload);
+      const index = state.chatRooms.findIndex(c => c.chatRoomId === chatRoomId);
```

Comment:
I thought we were changing the chat rooms to not be an array but an object with chatRoomId keys exactly so we don't have to do this findIndex thing

Reply:
next PR can? If got time?Sorry

Reply:
I  will put it in the issue tracker

Reply:
But again when a new message comes in, we sorting according to the attribute createdAt.

Reply:
Oh ya we discussed that. I think can add comment for that then haha